src/haddock/libs/libmpi.py

```diff
@@ -0,0 +1,46 @@
+"""Module in charge of MPI execution of tasks."""
+import pickle
+import shlex
+import subprocess
+import sys
+from pathlib import Path
+
+from haddock import log
+
+
+class MPIScheduler:
+    """Schedules tasks to be executed via MPI."""
+
+    def __init__(self, tasks, ncores=None):
+        self.tasks = tasks
+        self.cwd = Path.cwd()
+        self.ncores = ncores
+
+    def run(self):
+        """Send it to the haddock3-mpi runner."""
+        pkl_tasks = self._pickle_tasks()
+        cmd = f"mpirun -np {self.ncores} haddock3-mpi {pkl_tasks}"
+        log.debug(f"MPI cmd is {cmd}")
+
+        log.info(
+            f"Executing tasks with the haddock3-mpi runner using "
+            f"{self.ncores} processors..."
+            )
+        p = subprocess.run(
+            shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE
+            )
```

Comment:
Do you need to remove the `pkl_tasks` from disk afterwards?

Reply:
nope